.github/workflows/draw-zmk.yml

```
@@ -99,6 +104,37 @@ jobs:
         if: inputs.install_branch != ''
         run: python3 -m pip install "git+${{ inputs.install_repo }}@${{ inputs.install_branch }}"
 
+      - name: Install west
+        if: inputs.west_config_path != ''
+        run: python3 -m pip install west
+
+      - name: Cache west modules
+        if: inputs.west_config_path != ''
+        uses: actions/cache@v3.0.11
+        continue-on-error: true
+        env:
+          cache_name: cache-zephyr-${{ env.zephyr_version }}-modules
+        with:
+          path: |
+            modules/
+            tools/
+            zephyr/
+            bootloader/
+            zmk/
+          key: ${{ runner.os }}-build-${{ env.cache_name }}-${{ hashFiles('**/west.yml', '**/build.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache_name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
```

```suggestion
        with:
          path: |
            modules/
            tools/
            zephyr/
            bootloader/
            zmk/
          key: ${{ runner.os }}-draw-${{ hashFiles('**/west.yml') }}
          restore-keys: |
            ${{ runner.os }}-draw-
            ${{ runner.os }}-
```

We don't define a `zephyr_version` envvar before here, and it is probably not too necessary for us anyway.

Also simplified a bit to only hash west.yml and get rid of the `cache_name` altogether.

---

While it was copy pasted, leaving this as the `build` cache was mostly intentional. With this, it shared the same cache as the build pipeline. Which seems more optimal as if one already updated the cache, the other can reuse it. Though your `zephyr_version` got me wondering if i missed something.

Interestingly in the [official zmk workflow](https://github.com/zmkfirmware/zmk/blob/main/.github/workflows/build-user-config.yml), i also don't see where this variable is actually set 🤔 

Looking at my [latest build workflow](https://github.com/stijnveenman/zmk-config/actions/runs/9934940516/job/27440182067#step:6:36), it also had no zephyr version set. So this indeed shares the same cache and behaviour as the official workflow. 

 Updated your other 2 remarks, happy to hear your opinion on this one.

---

Reusing ZMK cache is a good point. 

> Interestingly in the [official zmk workflow](https://github.com/zmkfirmware/zmk/blob/main/.github/workflows/build-user-config.yml), i also don't see where this variable is actually set 🤔

I found it in the Docker image: https://github.com/zmkfirmware/zmk-docker/blob/3.5-branch/Dockerfile#L8. I think using it would be overkill here, but maybe we can hardcode a `zephyr_version` under `env:` and I could manually bump it as ZMK upgrades Zephyr versions. What do you think?

---

I don't think making it an input is necessary, we have a lot of inputs as is 😅 I think I'll be quick enough to bump it, and if not, an older cache doesn't break anything.

---

Good point, i've added `3.5.0` as a hardcoded version.